app/controllers/admin/users_controller.rb

```diff
@@ -1,6 +1,7 @@
 class Admin::UsersController < ApplicationController
   before_action :authorize_admin!
   before_action :authorize_super_admin!, only: [ :update, :destroy ]
+  after_action :remove_user_locks, only: [:destroy]
```

Comment:
Locks are using ```soft_delete!``` as well. 
And dependent can't point to :soft_delete! ```[:destroy, :delete_all, :nullify, :restrict_with_error, :restrict_with_exception]```
I don't think it is reasonable to override any of these methods, as it would be strange to modify the expected behaviour of ActiveRecord. It might be better to stay with the callback, moved into a concern.